typescript/ws-protocol/src/FoxgloveClient.ts

```diff
@@ -173,6 +192,31 @@ export default class FoxgloveClient {
     this.ws.send(payload);
   }
 
+  sendServiceRequest(request: ServiceCallRequest): void {
+    const utf8Encode = new TextEncoder();
+    const encoding = utf8Encode.encode(request.encoding);
+    const payload = new Uint8Array(1 + 4 + 4 + 4 + encoding.length + request.data.byteLength);
+    const view = new DataView(payload.buffer, payload.byteOffset, payload.byteLength);
+    let offset = 0;
+    view.setUint8(offset, ClientBinaryOpcode.SERVICE_CALL_REQUEST);
+    offset += 1;
+    view.setUint32(offset, request.serviceId, true);
+    offset += 4;
+    view.setUint32(offset, request.callId, true);
+    offset += 4;
+    view.setUint32(offset, request.encoding.length, true);
+    offset += 4;
+    payload.set(encoding, offset);
+    offset += encoding.length;
+    const data = new Uint8Array(
+      request.data.buffer,
+      request.data.byteOffset,
+      request.data.byteLength,
+    );
+    payload.set(data, offset);
```

Comment:
@jhurliman is there a better way to do that? I feel like we have to do to many Uint8Array / Dataview conversions

Reply:
This all looks right, unless `request.data` is already a Uint8Array. These conversions are lightweight